### PR TITLE
settings ui changed

### LINKED ISSUE #281: UI improvement for settings page
**UI improvement for settings page**

**To make the UI more user friendly, include some minimal icons and tweak up the whole user experience.**

**Would you like to work on the issue?**
**Yes , I would love to work on the issue.**

### PR DESCRIPTION
**Pull request by mistervaibhav**
NOTE: Please review the pull request process before opening your first PR: https://github.com/WikiEducationFoundation/apps-android-wikiedudashboard/blob/master/CONTRIBUTING.md#pull-request-process

## What this PR does
Fixes #281 

The UI of the settings page is hereby changed with some improvements in the user experience of clickable items, the logout button is more user friendly than ever and the position of the logo, title of the app and its version is changed such that it appears at the top of the activity.

## Screenshots
Before:
IMG-20200107-WA0000
After:
Screenshot (469)

### CODE REVIEW

**Comment by mistervaibhav** [comment]
Everything looks good however I am not sure why the build is failing, could you please help me out?

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -44,4 +44,19 @@
     <dimen name="dimens_168dp">168dp</dimen>
     <dimen name="dimens_200dp">200dp</dimen>
 
+    <dimen name="spacing_xsmall">2dp</dimen>
```
Comment (ujjwalagrawal17) [inline app/src/main/res/values/dimens.xml]: Please try to follow the exisiting naming convention (_xdp suffix.)and create a new issue if you suggest us to follow a better naming convention for dimensions. Add only dimensions that are needed. If possible, we will try to remove all the odd dimensions (3dp, 5dp,..) and use even dimensions (multiple of 4 in our code).

File: app/src/main/res/values/strings.xml [hunk 2]
```diff
@@ -83,9 +83,12 @@
     <string name="privacy_policy">Privacy Policy</string>
     <string name="terms_and_conditions">Terms and Conditions</string>
     <string name="logout">Logout</string>
-    <string name="version_0_01">Version 0.01</string>
+    <string name="version_0_01"> 0.01</string>
+
     <string name="license_link">License link</string>
-    <string name="total_impact_made_by_student">Total impact made by %1$!\'s students</string>
-    <string name="total_impact_made_as_student">Total impact made by %1$!\ as a student</string>
+    <string name="licence">Licence</string>
```
Comment (ujjwalagrawal17) [inline app/src/main/res/values/strings.xml]: licence  -> license

**Comment by ujjwalagrawal17** [comment]
Hi, Thanks for the PR! I will need to check if the build is passing in master or not. In the meantime, please look at the changes I requested.

File: app/src/main/res/values/dimens.xml (hunk 1 above)
Comment (mistervaibhav) [reply app/src/main/res/values/dimens.xml]: Okay , I have corrected those strings accordingly.

File: app/src/main/res/values/strings.xml (hunk 2 above)
Comment (mistervaibhav) [reply app/src/main/res/values/strings.xml]: I have corrected the typos.

**Comment by ujjwalagrawal17** [comment]
Can you also use constraint layout instead of nesting Linear Layouts?

**Comment by mistervaibhav** [comment]
> 
> 
> Can you also use constraint layout instead of nesting Linear Layouts?

**Comment by mistervaibhav** [comment]
> 
> 
> Can you also use constraint layout instead of nesting Linear Layouts?

For sure .

**Comment by mistervaibhav** [comment]
Hi @ujjwalagrawal17 , I have made all the changes that were suggested by you .

**Comment by ujjwalagrawal17** [comment]
@ragesoss which UI do you think is better?

**Comment by ujjwalagrawal17** [comment]
We still have nested linear layouts inside cardViews. Of we can remove them, it will be great.

**Comment by ragesoss** [comment]
They both look good, but I think the 'after' version is nicer. I like the icons and the extra info.

**Comment by mistervaibhav** [comment]
> We still have nested linear layouts inside cardViews. Of we can remove them, it will be great.

Sure thing @ujjwalagrawal17  I will work on it.

**Comment by gconnect** [comment]
> NOTE: Please review the pull request process before opening your first PR: https://github.com/WikiEducationFoundation/apps-android-wikiedudashboard/blob/master/CONTRIBUTING.md#pull-request-process
> 
> ## What this PR does
> Fixes #281
> 
> The UI of the settings page is hereby changed with some improvements in the user experience of clickable items, the logout button is more user friendly than ever and the position of the logo, title of the app and its version is changed such that it appears at the top of the activity.
> 
> ## Screenshots
> Before:
> IMG-20200107-WA0000
> After:
> Screenshot (469)

The latest design looks better. But it will be nice if you use a lighter shade of black for the icons and logout text

**Comment by ujjwalagrawal17** [comment]
@mistervaibhav can you change the button colours too as suggested by @gconnect ?

**Comment by mistervaibhav** [comment]
> 
> 
> @mistervaibhav can you change the button colours too as suggested by @gconnect ?

yeah sure @ujjwalagrawal17 ,  i will try different shades of black and try to sooth the UI as much as possible, meanwhile @gconnect  it would be great if you could suggest some shades , like I was experimenting with licorice (#1a1110) , Davy's gray (#555555) which is also the color of textViews, Outer Space (#414A4C ) and some other similar colors.

**Comment by ujjwalagrawal17** [comment]
@mistervaibhav as suggested by @gconnect we will like

#808080 - text.
#FB6E77 - logout button.

**Comment by mistervaibhav** [comment]
Screenshot (471)
@ujjwalagrawal17 @gconnect , I have used a new logout icon and changed the color of the text accordingly. I have also used lighter shades for other icons as well .
(ignore the dark line of the left side of the snapshot) I will also upload the preview with these changes #808080 - text.
#FB6E77 - logout button 
and it would become easier to compare and decide the better alternative. Currently, i am replacing the linear layouts so i would spend a little more time.
Also ,do give your comments on this particular preview.

**Comment by mistervaibhav** [comment]
Now this is how the UI looks....

Screenshot_20200109-122527

and when we scroll down a little ....

Screenshot_20200109-122536

@ujjwalagrawal17 
I have eliminated all the linear layouts as per your recommendations.

@gconnect  
I have changed the colors of all the icons accordingly and used a new logout icon and the Logout text is in accordance with the icon.
Kindly give your feedback and suggestions and let me know if I am required to make some more changes and modifications.
Thank you for your time.

**Comment by gconnect** [comment]
The icons on the left are still looking dark can it be lighter than that. The logout icon u used before was better just change the color of the icon

**Comment by mistervaibhav** [comment]
> 
> 
> The icons on the left are still looking dark can it be lighter than that. The logout icon u used before was better just change the color of the icon

@gconnect  Can you recommend any color for the icons ??

**Comment by mistervaibhav** [comment]
The logout issues as been resolved I believe . 

Screenshot_20200109-210000

@gconnect  I have tried using various colors for the icons and presented what i believed to be the best , and I think your suggestions would make it even better . I will update them as soon as you recommend them .

**Comment by ujjwalagrawal17** [comment]
Can you also share a logout button, logout icon with #FB6E77 color?

**Comment by mistervaibhav** [comment]
> 
> 
> Can you also share a logout button, logout icon with #FB6E77 color?

Here it is 
with #FB6E77 color.
Screenshot_20200109-214209

**Comment by gconnect** [comment]
> > The icons on the left are still looking dark can it be lighter than that. The logout icon u used before was better just change the color of the icon
> 
> @gconnect Can you recommend any color for the icons ??

#BFBFBF

**Comment by mistervaibhav** [comment]
photo_2020-01-10_20-39-37
photo_2020-01-10_20-39-41
@ujjwalagrawal17 @gconnect Have a look.

**Comment by gconnect** [comment]
It looks better now. Nice job!

**Comment by ujjwalagrawal17** [comment]
Thanks @mistervaibhav

**Comment by ujjwalagrawal17** [comment]
Nice Job 👍

**Pull request merged**